opennms-base-assembly/src/main/filtered/etc/syslog/NetgearProsafeSmartSwitch.syslog.xml

```diff
@@ -0,0 +1,218 @@
+            <uei>uei.opennms.org/vendor/netgear/syslog/prosafesmartswitch/Warning/EnergyEfficientEthernetStateTrue</uei>
+            <parameter-assignment matching-group="1" parameter-name="portName" />
+        </ueiMatch>
+        <ueiMatch>
+            <facility>local7</facility>
+            <severity>Warning</severity>
+            <process-match expression="^%NT_GREEN-W-EeeLldpMultiNeighbours$" />
+            <match type="regex" expression="^Multiple LLDP neighbours on port (.*) - EEE operational state is FALSE$" />
+            <uei>uei.opennms.org/vendor/netgear/syslog/prosafesmartswitch/Warning/EnergyEfficientEthernetStateFalse</uei>
+            <parameter-assignment matching-group="1" parameter-name="portName" />
+        </ueiMatch>
+        <ueiMatch>
+            <facility>local7</facility>
+            <severity>Warning</severity>
+            <process-match expression="^%STP-W-PORTSTATUS$" />
+            <match type="regex" expression="^(.*) of instance ([0-9.]+): STP status Forwarding$" />
+            <uei>uei.opennms.org/vendor/netgear/syslog/prosafesmartswitch/Warning/PortSTPStatusForwarding</uei>
+            <parameter-assignment matching-group="1" parameter-name="portName" />
+            <parameter-assignment matching-group="2" parameter-name="instanceNumber" />
+        </ueiMatch>
+        <ueiMatch>
+            <facility>local7</facility>
+            <severity>Warning</severity>
+            <process-match expression="^%BOOTP_DHCP_CL-W-FLNMEEMPTY$" />
+            <match type="substr" expression="No option 67 in the DHCP packet. Unable to start autoconfiguration." />
+            <uei>uei.opennms.org/vendor/netgear/syslog/prosafesmartswitch/Warning/UnableToStartDHCPAutoconfiguration</uei>
+        </ueiMatch>
+        <ueiMatch>
+            <facility>local7</facility>
+            <severity>Warning</severity>
+            <process-match expression="^%LINK-W-Down$" />
+            <match type="regex" expression="^(.*)$" />
+            <uei>uei.opennms.org/vendor/netgear/syslog/prosafesmartswitch/Warning/PortStatusLinkDown</uei>
+            <parameter-assignment matching-group="1" parameter-name="portName" />
+        </ueiMatch>
+        <ueiMatch>
+            <facility>local7</facility>
+            <severity>Warning</severity>
+            <process-match expression="^%AAA-W-REJECT$" />
+            <match type="regex" expression="^New http connection for user (.+), source ([0-9.]+) destination ([0-9.]+)  REJECTED$" /> <!-- two spaces needed -->
+            <uei>uei.opennms.org/vendor/netgear/syslog/prosafesmartswitch/Warning/WebConsoleLoginFailure</uei>
+            <parameter-assignment matching-group="1" parameter-name="userName" />
+            <parameter-assignment matching-group="2" parameter-name="sourceIP" />
+            <parameter-assignment matching-group="3" parameter-name="destinationIP" />
+        </ueiMatch>
+        <!-- catch-all event for other Warning syslog messages -->
```

Comment:
Hi Jeff, thanks very much for your feedback! I will revise as per your comments.

 

Re: too generic comment below

Just to confirm you are referring to the syslog events where I have defined “catch-all event for other xxx syslog messages” sections?

 

If so, unfortunately Netgear does not provide any documentation on what their possible syslog messages are (unlike their SNMP traps, where they provide a MIB file). I’ve just been recording the syslog messages based on actual experience. I’ll remove the generic syslog events to avoid false positives from other devices.

 

Regards,

Tm

 

 

From: Jeff Gehlbach [mailto:notifications@github.com] 
Sent: 11 February, 2016 11:17
To: OpenNMS/opennms <opennms@noreply.github.com>
Cc: Tim Miller Dyck <tim@millerdyck.org>
Subject: Re: [opennms] NMS-8077: Add Netgear ProSafe Smart Switch SNMP trap and syslog definitions and events (#550)

 

In opennms-base-assembly/src/main/filtered/etc/syslog/NetgearProsafeSmartSwitch.syslog.xml <https://github.com/OpenNMS/opennms/pull/550#discussion_r52624024> :

> +            <process-match expression="^%LINK-W-Down$" />
> +            <match type="regex" expression="^(.*)$" />
> +            <uei>uei.opennms.org/vendor/netgear/syslog/prosafesmartswitch/Warning/PortStatusLinkDown</uei>
> +            <parameter-assignment matching-group="1" parameter-name="portName" />
> +        </ueiMatch>
> +        <ueiMatch>
> +            <facility>local7</facility>
> +            <severity>Warning</severity>
> +            <process-match expression="^%AAA-W-REJECT$" />
> +            <match type="regex" expression="^New http connection for user (.+), source ([0-9.]+) destination ([0-9.]+)  REJECTED$" /> <!-- two spaces needed -->
> +            <uei>uei.opennms.org/vendor/netgear/syslog/prosafesmartswitch/Warning/WebConsoleLoginFailure</uei>
> +            <parameter-assignment matching-group="1" parameter-name="userName" />
> +            <parameter-assignment matching-group="2" parameter-name="sourceIP" />
> +            <parameter-assignment matching-group="3" parameter-name="destinationIP" />
> +        </ueiMatch>
> +        <!-- catch-all event for other Warning syslog messages -->

This definition is too generic. It is bound to match messages from non-Netgear devices. Suggest making it more specific or removing it altogether.

—
Reply to this email directly or view it on GitHub <https://github.com/OpenNMS/opennms/pull/550/files#r52624024> .  <https://github.com/notifications/beacon/AGRloh2COkI98mMd9XnfPS81nCg8RLR1ks5pjKuZgaJpZM4HHLW4.gif>